### PR TITLE
3168 mimikatz collector settings

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3168 .

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [ ] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "required": ["exclude_username_prefix"],
+    "properties": {
+        "exclude_username_prefix": {
+            "title": "Excluded username prefix",
+            "description": "A username prefix for which Mimikatz will not collect credentials.",
```
Comment (mssalvatore) [inline monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json:6]:
```suggestion
            "description": "Mimikatz will not collect credentials for any user whos username starts with this prefix.",
```

File: monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "required": ["exclude_username_prefix"],
+    "properties": {
```
Comment (mssalvatore) [inline monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json:3]: We decided not to allow the user to select credential types?

Comment (ilija-lazoroski) [reply monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json:3]: Nope, but we a tightly coupled to how Pypykatz is naming the credential types. See [this](https://github.com/skelsec/pypykatz/blob/1222aa210745ae66fa4a309f58c701186a4dcd8f/pypykatz/pypykatz.py#L348)

File: monkey/agent_plugins/credentials_collectors/mimikatz/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+name: Mimikatz
+plugin_type: Credentials_Collector
+supported_operating_systems:
+  - linux
+  - windows
+target_operating_systems:
+  - windows
+title: Mimikatz Credentials Collector
+version: 1.0.0
+description: Collects credentials from Windows credential manager.
```
Comment (shreyamalviya) [inline monkey/agent_plugins/credentials_collectors/mimikatz/manifest.yaml:10]:
```suggestion
description: Collects credentials from Windows Credential Manager using Mimikatz.
```

**Review by mssalvatore — APPROVED** [review]

File: monkey/agent_plugins/credentials_collectors/mimikatz/src/mimikatz_options.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from pydantic import Field
+
+from common.base_models import InfectionMonkeyBaseModel
+
+
+class MimikatzOptions(InfectionMonkeyBaseModel):
+    excluded_username_prefix: str = Field(
+        default="somenewuser",
+        description="Mimikatz will not collect credentials for any user whos username"
```
Comment (shreyamalviya) [inline monkey/agent_plugins/credentials_collectors/mimikatz/src/mimikatz_options.py:9]:
```suggestion
        description="Mimikatz will not collect credentials for any user whose username"
```

File: monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+        "exclude_username_prefix": {
+            "title": "Excluded username prefix",
+            "description": "A username prefix for which Mimikatz will not collect credentials.",
+            "type": "string",
+            "default": "somenewuser"
```
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/agent_plugins/credentials_collectors/mimikatz/config-schema.json:8]: We already have it hardcoded in the monkey model. https://github.com/guardicore/monkey/blob/5e8988d246e51e96ca41a230f91080f2867fbae4/monkey/infection_monkey/model/__init__.py#L6

**Pull request merged**